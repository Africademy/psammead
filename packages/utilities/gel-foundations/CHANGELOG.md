# Gel Foundations Changelog

| Version | Description |
|---------|-------------|
| 0.1.0   | [PR#XX](https://github.com/BBC-News/psammead/pull/XX) Create initial package, pulled in from gel-constants and gel-foundations-styled-components packages. |
