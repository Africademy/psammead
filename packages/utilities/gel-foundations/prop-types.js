module.exports = require('./dist/prop-types'); // eslint-disable-line import/no-unresolved
