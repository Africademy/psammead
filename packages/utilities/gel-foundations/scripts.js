/* eslint-disable global-require */
module.exports = {
  news: require('./dist/scripts/news').default,
  arabic: require('./dist/scripts/arabic').default,
  burmese: require('./dist/scripts/burmese').default,
  cjkThai: require('./dist/scripts/cjk_thai').default,
  hindiNepaliBengali: require('./dist/scripts/hindi_nepali_bengali').default,
  latin: require('./dist/scripts/arabic').default,
  mundo: require('./dist/scripts/arabic').default,
  sinhala: require('./dist/scripts/sinhala').default,
  tamil: require('./dist/scripts/tamil').default,
};
