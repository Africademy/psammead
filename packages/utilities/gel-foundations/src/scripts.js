/* eslint-disable global-require */
module.exports = {
  latin: require('../dist/scripts/latin-and-cyrillic').default,
  cyrillic: require('../dist/scripts/latin-and-cyrillic').default,
  latinDiacritics: require('../dist/scripts/latin-with-diacritics').default,
  arabic: require('../dist/scripts/arabic').default,
  arabicPashto: require('../dist/scripts/arabic-pashto').default,
  bengali: require('../dist/scripts/bengali').default,
  burmese: require('../dist/scripts/burmese').default,
  hindi: require('../dist/scripts/devanagari-and-gurmukhi').default,
  nepali: require('../dist/scripts/devanagari-and-gurmukhi').default,
  sinhalese: require('../dist/scripts/sinhalese').default,
  tamil: require('../dist/scripts/tamil').default,
  thai: require('../dist/scripts/thai').default,
  chinese: require('../dist/scripts/no-ascenders-or-descenders').default,
  korean: require('../dist/scripts/no-ascenders-or-descenders').default,
  ethiopic: require('../dist/scripts/no-ascenders-or-descenders').default,
};
