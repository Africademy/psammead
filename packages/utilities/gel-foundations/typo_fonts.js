module.exports = require('./dist/typo_fonts'); // eslint-disable-line import/no-unresolved
