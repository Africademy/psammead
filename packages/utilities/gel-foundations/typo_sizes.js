module.exports = require('./dist/typo_sizes'); // eslint-disable-line import/no-unresolved
