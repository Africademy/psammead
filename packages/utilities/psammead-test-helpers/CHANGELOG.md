# Psammead Test Helpers Changelog

| Version | Description |
|---------|-------------|
| 0.1.2   | [PR#255](https://github.com/BBC-News/psammead/pull/255) Bumps package version to fix what appears to be an erroneously-published version. |
| 0.1.2   | [PR#212](https://github.com/BBC-News/psammead/pull/212) Update package description and README. |
| 0.1.1   | [PR#173](https://github.com/BBC-News/psammead/pull/173) Update PRs welcome link |
| 0.1.0   | [PR#80](https://github.com/BBC-News/psammead/pull/80) Create initial package, pulled in from [simorgh](https://github.com/BBC-News/simorgh). |
