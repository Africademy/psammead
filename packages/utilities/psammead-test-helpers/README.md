# psammead-test-helpers &middot; [![GitHub license](https://img.shields.io/badge/license-Apache%202.0-blue.svg)](https://github.com/BBC-News/psammead/blob/latest/LICENSE) [![npm version](https://img.shields.io/npm/v/@bbc/psammead-test-helpers.svg)](https://www.npmjs.com/package/@bbc/psammead-test-helpers) [![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg)](https://reactjs.org/docs/how-to-contribute.html#your-first-pull-request)

This package is a collection helper methods used for snapshot testing, with support for styled components.

## Installation

```jsx
npm install @bbc/psammead-test-helpers --save
```

## Usage

```jsx
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';

shouldMatchSnapshot(
  'should render correctly',
  <h1>Hello World</h1>,
);
```

## Roadmap

There is currently a plan to migrate from `react-test-renderer` to `enzyme` as it provides a nicer higher-level API over the same functionality, as well as having support for more granular unit tests.

## Contributing

Psammead is completely open source. We are grateful for any contributions, whether they be new components, bug fixes or general improvements. Please see our primary contributing guide which can be found at [the root of the Psammead respository](https://github.com/BBC-News/psammead/blob/latest/CONTRIBUTING.md).

### [Code of Conduct](https://github.com/BBC-News/psammead/blob/latest/CODE_OF_CONDUCT.md)

We welcome feedback and help on this work. By participating in this project, you agree to abide by the [code of conduct](https://github.com/BBC-News/psammead/blob/latest/CODE_OF_CONDUCT.md). Please take a moment to read it.

### License

Psammead is [Apache 2.0 licensed](https://github.com/BBC-News/psammead/blob/latest/LICENSE).
