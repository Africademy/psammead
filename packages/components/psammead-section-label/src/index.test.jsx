import React from 'react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import { latin, arabic } from '@bbc/gel-foundations/scripts';
import { SectionLabel, SectionLabelWithoutBar } from './index';

describe('SectionLabelWithoutBar', () => {
  describe('With title', () => {
    shouldMatchSnapshot(
      'should render correctly',
      <SectionLabelWithoutBar script={latin}>
        This is text in a SectionLabelWithoutBar.
      </SectionLabelWithoutBar>,
    );

    shouldMatchSnapshot(
      'should render correctly with explicit text direction',
      <SectionLabelWithoutBar script={latin}>
        This is text in a SectionLabelWithoutBar rendering in ltr mode.
      </SectionLabelWithoutBar>,
    );

    shouldMatchSnapshot(
      'should render correctly with arabic script typography values',
      <SectionLabelWithoutBar script={arabic} dir="rtl">
        بعض محتوى النص
      </SectionLabelWithoutBar>,
    );
  });

  describe('Without title', () => {
    shouldMatchSnapshot('should render correctly', <SectionLabelWithoutBar />);
  });
});

describe('SectionLabel', () => {
  describe('With title', () => {
    shouldMatchSnapshot(
      'should render correctly',
      <SectionLabel script={latin}>
        This is text in a SectionLabel.
      </SectionLabel>,
    );

    shouldMatchSnapshot(
      'should render correctly with explicit text direction',
      <SectionLabel script={latin}>
        This is text in a SectionLabel rendering in ltr mode.
      </SectionLabel>,
    );

    shouldMatchSnapshot(
      'should render correctly with arabic script typography values',
      <SectionLabel script={arabic} dir="rtl">
        بعض محتوى النص
      </SectionLabel>,
    );
  });

  describe('Without title', () => {
    shouldMatchSnapshot('should render correctly', <SectionLabel />);
  });
});
