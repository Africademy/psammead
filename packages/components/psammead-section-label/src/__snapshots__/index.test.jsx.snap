// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionLabel When hideSectionHeader is true should add styling to hide SectionLabel for all breakpoints 1`] = `
<div>
  <div
    class="SectionLabelWrapper-sc-1x4gwjz-1 doOlE"
  >
    <h2
      class="Heading-sc-1x4gwjz-2 eZofTG"
    >
      <span
        class="FlexColumn-wx5kdp-0 dgpsiS"
      >
        <span
          class="FlexRow-wx5kdp-2 dVBNyZ"
        >
          <span
            class="Title-wx5kdp-4 dfRrVv"
            dir="ltr"
            id="test-section-label"
          >
            This is the text in a SectionLabel
          </span>
        </span>
      </span>
    </h2>
  </div>
</div>
`;

exports[`SectionLabel With bar With linking title should render correctly 1`] = `
<div>
  <div
    class="SectionLabelWrapper-sc-1x4gwjz-1 hsirkD"
  >
    <div
      class="Bar-sc-1x4gwjz-0 jsgdXU"
    />
    <h2
      class="Heading-sc-1x4gwjz-2 eZofTG"
    >
      <a
        aria-labelledby="test-section-label"
        class="SectionLabelLink-wx5kdp-1 hlUWNS"
        href="/igbo/other-index"
      >
        <span
          class="FlexColumn-wx5kdp-0 dgpsiS"
        >
          <span
            class="FlexRow-wx5kdp-2 FlexTextRow-wx5kdp-3 jvseIt"
            role="text"
          >
            <span
              class="Title-wx5kdp-4 dfRrVv"
              dir="ltr"
              id="test-section-label"
            >
              This is text in a linking SectionLabel.
            </span>
            <span
              aria-hidden="true"
              class="IndexLinkCta-wx5kdp-5 lnsJfj"
              dir="ltr"
            >
              See All
            </span>
          </span>
        </span>
      </a>
    </h2>
  </div>
</div>
`;

exports[`SectionLabel With bar With linking title should render correctly with arabic script typography values 1`] = `
<div>
  <div
    class="SectionLabelWrapper-sc-1x4gwjz-1 hsirkD"
  >
    <div
      class="Bar-sc-1x4gwjz-0 kXOyLn"
    />
    <h2
      class="Heading-sc-1x4gwjz-2 eZofTG"
    >
      <a
        aria-labelledby="test-section-label"
        class="SectionLabelLink-wx5kdp-1 hlUWNS"
        href="/igbo/other-index"
      >
        <span
          class="FlexColumn-wx5kdp-0 dgpsiS"
        >
          <span
            class="FlexRow-wx5kdp-2 FlexTextRow-wx5kdp-3 jvseIt"
            role="text"
          >
            <span
              class="Title-wx5kdp-4 jJpioi"
              dir="rtl"
              id="test-section-label"
            >
              بعض محتوى النص
            </span>
            <span
              aria-hidden="true"
              class="IndexLinkCta-wx5kdp-5 gHULMu"
              dir="rtl"
            >
              See All
            </span>
          </span>
        </span>
      </a>
    </h2>
  </div>
</div>
`;

exports[`SectionLabel With bar With linking title should render correctly with explicit text direction 1`] = `
<div>
  <div
    class="SectionLabelWrapper-sc-1x4gwjz-1 hsirkD"
  >
    <div
      class="Bar-sc-1x4gwjz-0 jsgdXU"
    />
    <h2
      class="Heading-sc-1x4gwjz-2 eZofTG"
    >
      <a
        aria-labelledby="test-section-label"
        class="SectionLabelLink-wx5kdp-1 hlUWNS"
        href="/igbo/other-index"
      >
        <span
          class="FlexColumn-wx5kdp-0 dgpsiS"
        >
          <span
            class="FlexRow-wx5kdp-2 FlexTextRow-wx5kdp-3 jvseIt"
            role="text"
          >
            <span
              class="Title-wx5kdp-4 dfRrVv"
              dir="ltr"
              id="test-section-label"
            >
              This is text in a SectionLabel rendering in ltr mode.
            </span>
            <span
              aria-hidden="true"
              class="IndexLinkCta-wx5kdp-5 lnsJfj"
              dir="ltr"
            >
              See All
            </span>
          </span>
        </span>
      </a>
    </h2>
  </div>
</div>
`;

exports[`SectionLabel With bar With linking title should render correctly with explicitly showing the bar 1`] = `
<div>
  <div
    class="SectionLabelWrapper-sc-1x4gwjz-1 hsirkD"
  >
    <div
      class="Bar-sc-1x4gwjz-0 jsgdXU"
    />
    <h2
      class="Heading-sc-1x4gwjz-2 eZofTG"
    >
      <a
        aria-labelledby="test-section-label"
        class="SectionLabelLink-wx5kdp-1 hlUWNS"
        href="/igbo/other-index"
      >
        <span
          class="FlexColumn-wx5kdp-0 dgpsiS"
        >
          <span
            class="FlexRow-wx5kdp-2 FlexTextRow-wx5kdp-3 jvseIt"
            role="text"
          >
            <span
              class="Title-wx5kdp-4 dfRrVv"
              dir="ltr"
              id="test-section-label"
            >
              This is text in a SectionLabel, and there is a bar over to the right
            </span>
            <span
              aria-hidden="true"
              class="IndexLinkCta-wx5kdp-5 lnsJfj"
              dir="ltr"
            >
              See All
            </span>
          </span>
        </span>
      </a>
    </h2>
  </div>
</div>
`;

exports[`SectionLabel With bar With plain title should render correctly 1`] = `
<div>
  <div
    class="SectionLabelWrapper-sc-1x4gwjz-1 hsirkD"
  >
    <div
      class="Bar-sc-1x4gwjz-0 jsgdXU"
    />
    <h2
      class="Heading-sc-1x4gwjz-2 eZofTG"
    >
      <span
        class="FlexColumn-wx5kdp-0 dgpsiS"
      >
        <span
          class="FlexRow-wx5kdp-2 dVBNyZ"
        >
          <span
            class="Title-wx5kdp-4 dfRrVv"
            dir="ltr"
            id="test-section-label"
          >
            This is text in a SectionLabel.
          </span>
        </span>
      </span>
    </h2>
  </div>
</div>
`;

exports[`SectionLabel With bar With plain title should render correctly with arabic script typography values 1`] = `
<div>
  <div
    class="SectionLabelWrapper-sc-1x4gwjz-1 hsirkD"
  >
    <div
      class="Bar-sc-1x4gwjz-0 kXOyLn"
    />
    <h2
      class="Heading-sc-1x4gwjz-2 eZofTG"
    >
      <span
        class="FlexColumn-wx5kdp-0 dgpsiS"
      >
        <span
          class="FlexRow-wx5kdp-2 dVBNyZ"
        >
          <span
            class="Title-wx5kdp-4 jJpioi"
            dir="rtl"
            id="test-section-label"
          >
            بعض محتوى النص
          </span>
        </span>
      </span>
    </h2>
  </div>
</div>
`;

exports[`SectionLabel With bar With plain title should render correctly with explicit text direction 1`] = `
<div>
  <div
    class="SectionLabelWrapper-sc-1x4gwjz-1 hsirkD"
  >
    <div
      class="Bar-sc-1x4gwjz-0 jsgdXU"
    />
    <h2
      class="Heading-sc-1x4gwjz-2 eZofTG"
    >
      <span
        class="FlexColumn-wx5kdp-0 dgpsiS"
      >
        <span
          class="FlexRow-wx5kdp-2 dVBNyZ"
        >
          <span
            class="Title-wx5kdp-4 dfRrVv"
            dir="ltr"
            id="test-section-label"
          >
            This is text in a SectionLabel rendering in ltr mode.
          </span>
        </span>
      </span>
    </h2>
  </div>
</div>
`;

exports[`SectionLabel With bar With plain title should render correctly with explicitly showing the bar 1`] = `
<div>
  <div
    class="SectionLabelWrapper-sc-1x4gwjz-1 hsirkD"
  >
    <div
      class="Bar-sc-1x4gwjz-0 jsgdXU"
    />
    <h2
      class="Heading-sc-1x4gwjz-2 eZofTG"
    >
      <span
        class="FlexColumn-wx5kdp-0 dgpsiS"
      >
        <span
          class="FlexRow-wx5kdp-2 dVBNyZ"
        >
          <span
            class="Title-wx5kdp-4 dfRrVv"
            dir="ltr"
            id="test-section-label"
          >
            This is text in a SectionLabel, and there is a bar over to the right
          </span>
        </span>
      </span>
    </h2>
  </div>
</div>
`;

exports[`SectionLabel With heading overriden should render a strong element instead of an h2 1`] = `
<div>
  <div
    class="SectionLabelWrapper-sc-1x4gwjz-1 hsirkD"
  >
    <strong
      class="Heading-sc-1x4gwjz-2 eZofTG"
    >
      <span
        class="FlexColumn-wx5kdp-0 dgpsiS"
      >
        <span
          class="FlexRow-wx5kdp-2 dVBNyZ"
        >
          <span
            class="Title-wx5kdp-4 dfRrVv"
            dir="ltr"
            id="test-section-label"
          >
            This is text in a SectionLabel.
          </span>
        </span>
      </span>
    </strong>
  </div>
</div>
`;

exports[`SectionLabel Without bar With linking title should render correctly 1`] = `
<div>
  <div
    class="SectionLabelWrapper-sc-1x4gwjz-1 hsirkD"
  >
    <h2
      class="Heading-sc-1x4gwjz-2 eZofTG"
    >
      <a
        aria-labelledby="test-section-label"
        class="SectionLabelLink-wx5kdp-1 hlUWNS"
        href="/igbo/other-index"
      >
        <span
          class="FlexColumn-wx5kdp-0 dgpsiS"
        >
          <span
            class="FlexRow-wx5kdp-2 FlexTextRow-wx5kdp-3 jvseIt"
            role="text"
          >
            <span
              class="Title-wx5kdp-4 dfRrVv"
              dir="ltr"
              id="test-section-label"
            >
              This is text in a SectionLabel.
            </span>
            <span
              aria-hidden="true"
              class="IndexLinkCta-wx5kdp-5 lnsJfj"
              dir="ltr"
            >
              See All
            </span>
          </span>
        </span>
      </a>
    </h2>
  </div>
</div>
`;

exports[`SectionLabel Without bar With linking title should render correctly with arabic script typography values 1`] = `
<div>
  <div
    class="SectionLabelWrapper-sc-1x4gwjz-1 hsirkD"
  >
    <h2
      class="Heading-sc-1x4gwjz-2 eZofTG"
    >
      <a
        aria-labelledby="test-section-label"
        class="SectionLabelLink-wx5kdp-1 hlUWNS"
        href="/igbo/other-index"
      >
        <span
          class="FlexColumn-wx5kdp-0 dgpsiS"
        >
          <span
            class="FlexRow-wx5kdp-2 FlexTextRow-wx5kdp-3 jvseIt"
            role="text"
          >
            <span
              class="Title-wx5kdp-4 jJpioi"
              dir="rtl"
              id="test-section-label"
            >
              بعض محتوى النص
            </span>
            <span
              aria-hidden="true"
              class="IndexLinkCta-wx5kdp-5 gHULMu"
              dir="rtl"
            >
              See All
            </span>
          </span>
        </span>
      </a>
    </h2>
  </div>
</div>
`;

exports[`SectionLabel Without bar With linking title should render correctly with explicit text direction 1`] = `
<div>
  <div
    class="SectionLabelWrapper-sc-1x4gwjz-1 hsirkD"
  >
    <h2
      class="Heading-sc-1x4gwjz-2 eZofTG"
    >
      <a
        aria-labelledby="test-section-label"
        class="SectionLabelLink-wx5kdp-1 hlUWNS"
        href="/igbo/other-index"
      >
        <span
          class="FlexColumn-wx5kdp-0 dgpsiS"
        >
          <span
            class="FlexRow-wx5kdp-2 FlexTextRow-wx5kdp-3 jvseIt"
            role="text"
          >
            <span
              class="Title-wx5kdp-4 dfRrVv"
              dir="ltr"
              id="test-section-label"
            >
              This is text in a SectionLabel rendering in ltr mode.
            </span>
            <span
              aria-hidden="true"
              class="IndexLinkCta-wx5kdp-5 lnsJfj"
              dir="ltr"
            >
              See All
            </span>
          </span>
        </span>
      </a>
    </h2>
  </div>
</div>
`;

exports[`SectionLabel Without bar With plain title should render correctly 1`] = `
<div>
  <div
    class="SectionLabelWrapper-sc-1x4gwjz-1 hsirkD"
  >
    <h2
      class="Heading-sc-1x4gwjz-2 eZofTG"
    >
      <span
        class="FlexColumn-wx5kdp-0 dgpsiS"
      >
        <span
          class="FlexRow-wx5kdp-2 dVBNyZ"
        >
          <span
            class="Title-wx5kdp-4 dfRrVv"
            dir="ltr"
            id="test-section-label"
          >
            This is text in a SectionLabel.
          </span>
        </span>
      </span>
    </h2>
  </div>
</div>
`;

exports[`SectionLabel Without bar With plain title should render correctly with arabic script typography values 1`] = `
<div>
  <div
    class="SectionLabelWrapper-sc-1x4gwjz-1 hsirkD"
  >
    <h2
      class="Heading-sc-1x4gwjz-2 eZofTG"
    >
      <span
        class="FlexColumn-wx5kdp-0 dgpsiS"
      >
        <span
          class="FlexRow-wx5kdp-2 dVBNyZ"
        >
          <span
            class="Title-wx5kdp-4 jJpioi"
            dir="rtl"
            id="test-section-label"
          >
            بعض محتوى النص
          </span>
        </span>
      </span>
    </h2>
  </div>
</div>
`;

exports[`SectionLabel Without bar With plain title should render correctly with explicit text direction 1`] = `
<div>
  <div
    class="SectionLabelWrapper-sc-1x4gwjz-1 hsirkD"
  >
    <h2
      class="Heading-sc-1x4gwjz-2 eZofTG"
    >
      <span
        class="FlexColumn-wx5kdp-0 dgpsiS"
      >
        <span
          class="FlexRow-wx5kdp-2 dVBNyZ"
        >
          <span
            class="Title-wx5kdp-4 dfRrVv"
            dir="ltr"
            id="test-section-label"
          >
            This is text in a SectionLabel rendering in ltr mode.
          </span>
        </span>
      </span>
    </h2>
  </div>
</div>
`;
